src/main/resources/avro/readmethods.avdl

```diff
@@ -14,8 +14,10 @@ to `ReferenceSet`s containing that same `Reference`. If no reference is
 specified, all `ReadGroup`s must be aligned to the same `ReferenceSet`.
 */
 record SearchReadsRequest {
-  /** If nonempty, restrict this query to reads within the given readgroups. */
-  array<string> readGroupIds = [];
+  /**
+  The ReadGroup to search.
+  */
+  string readGroupId;
```

Comment:
I agree with Heng that the ability of servers to integrate over readgroups seems activity from the user point of view.  The ability to look 
at a (normally small) region of the genome in thousands of samples is a real issue at the moment, and will be more so going forwards, 
and if we had a server that dealt with it that would be a true win.  Where this is in the server/client world I don't know - I can imagine
wanting to integrate across multiple servers each with different repositories I have access to, so I can see some client side integration
as valuable.  But that leaves the question of what the real API is - is it the remote one across the network, or some user code accessible
API?  I think at the user coding level one should have integration across read groups as a basic operation.  As Jerome says, at the
user code level one also wants all the paging to be hidden and just to use iterators across sets of objects.

How is this handled in other APIs?  Is there a distinction between a user API and a low-level network API that can satisfy it, but which
is optimised for simplicity and performance?

Richard

> On 24 Mar 2015, at 20:41, Paul Grosu <notifications@github.com> wrote:
> 
> In src/main/resources/avro/readmethods.avdl <https://github.com/ga4gh/schemas/pull/253#discussion_r27070656>:
> 
> > @@ -14,8 +14,10 @@ to `ReferenceSet`s containing that same `Reference`. If no reference is
> >  specified, all `ReadGroup`s must be aligned to the same `ReferenceSet`.
> >  */
> >  record SearchReadsRequest {
> > -  /** If nonempty, restrict this query to reads within the given readgroups. */
> > -  array<string> readGroupIds = [];
> > +  /**
> > +  The ReadGroup to search.
> > +  */
> > +  string readGroupId;
> Hi David,
> 
> Welcome back :) I like the points your brig up and I agree that we need to find a good balance, so maybe we should try to have that discussion now. Let's take the Google Maps API, how much is performed on the server vs the client? The data throughput for today's analysis would be of a similar scale. Would one perform the Google Distance Matrix API <https://developers.google.com/maps/documentation/distancematrix/> on their client? Most likely not. So we want to keep petabyte sub-second response times in mind as this API and it's implementations with distributed repositories scales.
> 
> Regarding adapting the API by taking into account currently implemented versions, has pros and cons. On the pro-side it would tell us what is possible and what to tweak. On the con-side, we do not want it to limit us. I have dealt in the past with similar situations on very-large implementations, and one definitely does not want to migrate the data from one schema to another. It is a huge pain, which I could image it to be at Google if the schema changes significantly here. But let me ask you this. What if in the future we brainstorm a major re-write that would improve the overall implementation? Should we go forward, or should we stay away if the currently implemented versions of the API would break? I think the goal is to work through these versions (since we are still sub-1.0 on GA4GH API) until we find the optimal setup.
> 
> What do you think?
> 
> Thanks,
> Paul
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/ga4gh/schemas/pull/253/files#r27070656>.
> 




-- 
 The Wellcome Trust Sanger Institute is operated by Genome Research 
 Limited, a charity registered in England with number 1021457 and a 
 company registered in England with number 2742969, whose registered 
 office is 215 Euston Road, London, NW1 2BE.